webpush.go

```diff
@@ -243,10 +245,16 @@ func getHKDFKey(hkdf io.Reader, length int) ([]byte, error) {
 	return key, nil
 }
 
-func pad(payload *bytes.Buffer, maxPadLen int) {
+func pad(payload *bytes.Buffer, maxPadLen int) error {
 	payloadLen := payload.Len()
+	if payloadLen > maxPadLen {
+		return errors.New("payload is too large")
```

Comment:
Could you move this to a package level variable:

```go
var ErrMaxPadExceeded = errors.New("Payload has exceeded the maximum padding length")
```

Reply:
sure, we can have it as a package level error! The error message should, I think, stay lowercase to comply with https://github.com/golang/go/wiki/CodeReviewComments#error-strings . I'm also not sure mentioning padding in there is a good idea, since this is an internal detail the user has probably no idea about?